projects/collection-editor-library/src/lib/services/editor/editor.service.ts

```diff
@@ -159,20 +159,36 @@ export class EditorService {
     return formFields;
   }
 
-  updateCollection(collectionId, data?) {
+  updateCollection(collectionId, event?) {    
     let objType = this.configService.categoryConfig[this.editorConfig.config.objectType];
+    let url = this.configService.urlConFig.URLS[this.editorConfig.config.objectType];
+    let requestBody = {
+      request: { }
+    };
     objType = objType.toLowerCase();
-    const url = this.configService.urlConFig.URLS[this.editorConfig.config.objectType];
-    const fieldsObj = this.getFieldsToUpdate(collectionId);
-    const requestBody = {
-      request: {
-        [objType]: {
-          ...fieldsObj,
-          lastPublishedBy: this.editorConfig.context.user.id
+
+    if(event.button === 'sourcingApproveQuestion' || event.button === 'sourcingRejectQuestion') {
```

Comment:
Can you please explain this?

Reply:
`sourcingApproveQuestion` and `sourcingRejectQuestion` are events called when we "Approve" or "Reject" a question as a second level reviewer. These events are handled in question component by method `sourcingUpdate` where this `updateCollection` is called. By default the endpoint is mapped using `objectType` which is `question` in this case. But we want to update `questionset` in case of `sourcingApproveQuestion` or `sourcingRejectQuestion` that is why we added this condition to overwrite the default endpoint and request body to update the `questionset`